libs/datasets/timeseries.py

```diff
@@ -292,6 +282,62 @@ def load_csv(cls, path_or_buf: Union[pathlib.Path, TextIO]):
         return cls(df)
 
 
+@final
+@dataclass(frozen=True)
+class MultiRegionTimeseriesDataset:
```

Comment:
The timeseries.py file is getting a bit crowded, thoughts on when we should separate the existing classes into multiple files?

Reply:
I'd like to replace the last of TimeseriesDataset with the more static and specific OneRegion/MultiRegionTimeseriesDataset, shrinking this file back down a lot.